### ISSUE TITLE
Suggestion: a new rule `NoEmptyLinesOpeningClosingBracesOnMethods` - No empty lines on opening or closing braces in methods only

### ISSUE THREAD

**Issue by Pispa339** [swiftlang/swift-format#819]
The existing rule `NoEmptyLinesOpeningClosingBraces` is somewhat too much of a catch-all rule. I'd like to request a more fine-grained rule where empty lines on opening or closing braces would be disallowed for methods, but allowed for type declarations.

For example the following would be disallowed within the rule
```
func hello() {

    // code starts here
```
whereas the following would be allowed
```
class SomeClass {

    // Code starts here
```

**Comment by ahoppen** [comment]
Synced to Apple’s issue tracker as rdar://136754857

**Comment by allevato** [comment]
These are the kinds of extremely fine-grained customizations that I'm really hesitant to add. `NoEmptyLinesOpeningClosingBraces` was added because it allows the formatter to be *more* opinionated across the board (which I think is a good goal). What would be the benefit of allowing the blank line for types but not functions?

**Comment by 0xLucasMarcal** [comment]
@allevato Any plans on supporting custom rules?

**Comment by allevato** [comment]
If by "custom rules" you mean some sort of plugin architecture where users would write something like a .dylib that contained additional rules that swift-format would recognize, it would be very nice to support something like that but we'd have to figure out the architectural details for it.

**Comment by 0xLucasMarcal** [comment]
@allevato Yeah, that is exactly what I was imagining. I would love to contribute to the architectural design and the implementation of it. Let me know how I can best contribute to that with you :)
